### ISSUE TITLE
ValueTypes: Add support for ojdk_mh

### ISSUE THREAD

**Issue by tajila** [eclipse-openj9/openj9#13549]
The existing valuetypes JDK is based off of jdknext. This repo doesn't have the required VT support in method handles. We will need to start building off https://github.com/openjdk/valhalla instead of the head repo.

In the short term, we will need to change our tests to use `makeValueGeneric` helpers and `generate[Getter|Setter|Wither]Generic` helpers as they do not mention q-types. Anything that mentions q-types right now will not work with ojdk_mh.

**Comment by tajila** [comment]
This is blocking https://github.com/eclipse-openj9/openj9/issues/13547

**Comment by hangshao0** [comment]
In the middle of #13414, may take me some time to get to this.

**Comment by tajila** [comment]
Okay, @EricYangIBM can you take a look at this

**Comment by JasonFengJ9** [comment]
> Long term we will need to make a new repo https://github.com/ibmruntimes/openj9-openjdk-jdk.valuetypes which will be based off https://github.com/openjdk/valhalla

Is this `openj9-openjdk-jdk.valuetypes` going to be the repo `github.com/openjdk/valhalla` w/ OpenJ9 JCL patches just like `openj9-openjdk-jdk` which is `github.com/openjdk/jdk` w/ JCL patches?
It sounds another extension repo mirror/merging build from `github.com/openjdk/valhalla` regularly before `valhalla` is merged into `jdk` head stream, right?

**Comment by tajila** [comment]
> It sounds another extension repo mirror/merging build from github.com/openjdk/valhalla regularly before valhalla is merged into jdk head stream, right?

Yes, thats the idea. The closer we get to the release the more JCL changes we will need for VTs

**Comment by tajila** [comment]
We will also need to do something like this for Loom

**Comment by JasonFengJ9** [comment]
> Yes, thats the idea. The closer we get to the release the more JCL changes we will need for VTs
We will also need to do something like this for Loom

Sure, do we have some timelines set for these repos?

**Comment by JasonFengJ9** [comment]
> Sooner the better. It will require some work from releng side so I imagine we can get to this after the next couple releases.

Tracking this at `infrastructure/issues/6004`.
